cg/store/filters/status_order_filters.py

```diff
@@ -6,27 +6,37 @@
 from cg.store.models import Order
 
 
-def filter_orders_by_workflow(orders: Query, workflow: str, **kwargs) -> Query:
-    """Return orders filtered on workflow."""
+def filter_orders_by_workflow(orders: Query, workflow: str | None, **kwargs) -> Query:
     return orders.filter(Order.workflow == workflow) if workflow else orders
 
 
-def filter_orders_by_id(orders: Query, id: int, **kwargs) -> Query:
-    """Return orders filtered on id."""
-    return orders.filter(Order.id == id)
+def filter_orders_by_id(orders: Query, id: int | None, **kwargs) -> Query:
+    return orders.filter(Order.id == id) if id else orders
 
 
-def apply_order_filters(
-    filter_functions: list[Callable], orders: Query, id: int = None, workflow: str = None
-) -> Query:
-    """Apply filtering functions to the order queries and return filtered results."""
-    for filter_function in filter_functions:
-        orders: Query = filter_function(orders=orders, id=id, workflow=workflow)
-    return orders
+def filter_orders_by_ids(orders: Query, ids: list[int] | None, **kwargs) -> Query:
+    return orders.filter(Order.id.in_(ids)) if ids else orders
+
+
+def apply_limit(orders: Query, limit: int | None, **kwargs) -> Query:
+    return orders.limit(limit) if limit else orders
 
 
 class OrderFilter(Enum):
-    """Define order filter functions."""
+    BY_ID: Callable = filter_orders_by_id
+    BY_IDS: Callable = filter_orders_by_ids
+    BY_WORKFLOW: Callable = filter_orders_by_workflow
+    APPLY_LIMIT: Callable = apply_limit
 
-    FILTER_ORDERS_BY_ID: Callable = filter_orders_by_id
-    FILTER_ORDERS_BY_WORKFLOW: Callable = filter_orders_by_workflow
+
+def apply_order_filters(
+    filters: list[OrderFilter],
+    orders: Query,
+    id: int = None,
+    ids: list[int] = None,
```

Comment:
I understand the reasoning for it, but something feels off with having both id and ids as parameters. Is it nicer than just converting the sole id into a list and using ids in all cases?

Reply:
I think so!